ui-v2/app/templates/dc/acls/tokens/-fieldsets-legacy.hbs

```diff
@@ -15,7 +15,7 @@
 {{/if}}
         <label class="type-text">
             <span>Rules <a href="{{env 'CONSUL_DOCUMENTATION_URL'}}/guides/acl.html#rule-specification" rel="help noopener noreferrer" target="_blank">(HCL Format)</a></span>
-            {{code-editor class=(if item.error.Rules 'error') name='Rules' value=item.Rules onkeyup=(action 'change' 'Rules')}}
+            {{code-editor class=(if item.error.Rules 'error') name='Rules' syntax='hcl' value=item.Rules onkeyup=(action 'change' 'Rules')}}
```

Comment:
You mean an ember `alias` yeah? Do you know how that would help here? I thought an `alias` was to point one property to another?

The `toLowerCase` was just good hygiene really, I took the mode specs from the code-mirror source and the properties/'syntax names' are in uppercase, and I didn't want to change that, so I thought there would always be the slight chance of confusion as to whether to say `syntax="HCL"` or `syntax="hcl"`

Actually looking at it I could at least use `===` not `==`

Anyway, lemme know on the alias thing, maybe I've misunderstood what you meant.

Reply:
Oh I meant https://github.com/hashicorp/consul/pull/4866/files/98acdadaf1a3597bd3e36247b6e321aa3b1826c3#diff-c79b0ce21a1201a8516d52bd57f5404bR17 one of those things. I might be really confused.

Just hygienic regardless so LGTM.

Reply:
Ahh, of course (sorry just realized what you meant now), you're not confused, I am!

This will end up going in post 1.4.0, but if you are interested in the background (more or less), here goes:

So, that JSON blob is taken directly from the codemirror `modes.js` file (I think its called that). That file is massive and contains a load of stuff I'll never need (it has ALL the different modes it supports). So I took the bits I need from it, but I want to leave it as close the original source as possible.

The HCL part of the blob is actually the 'ruby' one from codemirror, I just changed the `name:` but kept it uppercase to be consistent to the other names. So essentially HCL 'mode' is actually 'ruby' mode. I chose the `syntax` name for the attribute as it made sense to me, and I didn't want to use `name` as that means something else in HTML and `mode` didn't seem quite right as this is `ruby` mode..but hcl :S

To choose the syntax/mode, it just looks through the JSON blob, looking for `'HCL'.toLowerCase()`. I could have just hardcoded the names in the JSON blob to be lowercase, but see above about wanting to change as little as possible.